pkgs/tools/misc/grub/pvgrub_image/default.nix

```diff
@@ -0,0 +1,42 @@
+{ stdenv, grub2_xen }:
+
+with stdenv.lib;
+let
+  efiSystemsBuild = {
+    "i686-linux".target = "i386";
+    "x86_64-linux".target = "x86_64";
+    "aarch64-linux".target = "aarch64";
+  };
+
+in (
+
+stdenv.mkDerivation rec {
+  name = "pvgrub-image";
+
+  configs = ./configs;
+
+  buildInputs = [ grub2_xen ];
+
+  buildCommand = ''
+    cp "${configs}"/* .
+    tar -cf memdisk.tar grub.cfg
+    # We include all modules except all_video.mod as otherwise grub will fail printing "no symbol table"
```

Comment:
The comment I wrote was based on my observation that pvgrub fails when the `all_video.mod` module is included. The page you linked talks about booting Xen itself using EFI, which requires compiling grub as an EFI loader. In that particular case the `all_video.mod` module seems to work, but that does not contradict what I wrote. Please let me know if I should add any clarification to the comment.